### PR TITLE
[fix #211] Improves split_name helper & adds tests

### LINKED ISSUE #211: imporve split_name helper & add tests
need to handle edge case of just a space being sent in the data

### PR DESCRIPTION
**Pull request by jpetto**

### CODE REVIEW

File: basket/news/utils.py [hunk 1]
```diff
@@ -583,3 +583,36 @@ def parse_newsletters(api_call_type, newsletters, cur_newsletters):
...
+    # try to make the final bit after the last space the last name
+    names = name.rsplit(None, 1)
+
+    if len(names) == 2:
+        first, last = names
+
+        # if last name is 'jr' or 'sr' and first name has a space in it, do
+        # more splitting
+        if ' ' in first and last.lower() in ['jr', 'sr']:
+            names = first.rsplit(None, 1)
```
Comment (pmac) [inline basket/news/utils.py]: style nit: you can just do:

```suggestion
            first, last = first.rsplit(None, 1)
```

**Comment by jpetto** [comment]
yeah, i didn't think that auto commit would work. fixing...

**Comment by jpetto** [comment]
okay, should be set!
